Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java

```diff
@@ -456,6 +481,101 @@ public int compare(String text1, String text2) {
         }
     }
 
+    private void importData(Uri uri) {
+        try {
+            String exportContents = readFile(uri);
+            switch (getFileExtension(uri)) {
```

Comment:
I tried confirming the return values for `getType(uri)` and didn't see anything helpful. did you see a list? I saw that it does chop off the first part of the mime type.

we probably want to add unit tests to cover this function for the expected file types we will support importing. can it really suggest `md` as the extension? I guess if it detects plaintext _and_ has `.md`?

this is a nice update from before though. I like that there's some leeway if someone has a file with a missing or wrong file extension.

Reply:
Maybe I misunderstood, but I guess it works a little differently. It seems to me that it makes no difference to us what extension the file is, except for json, of course. All the rest will definitely be text (this is exactly what mimes are for), because I set mime types `application/json` and `text/*`, which includes txt, md, html and other, but there's no difference for us between them.

Reply:
Here's a list (section "common examples"): en.wikipedia.org/wiki/Media_type

Reply:
my point was that I don't see how it's possible here to ever match on `"md"`, unless Android is making a guess on that which I don't think it would?

there is a distinction that's possible, though we could handle all `text/plain` files the same way, but that distinction is whether or not we toggle markdown support in the note. if we happen to know it's a `.md` file _and_ it's plaintext then why not go ahead and import it with markdown enabled. otherwise, if it's plaintext import it just as raw text.

Reply:
Ah, I understand, I apologize for what you had to explain. However, unfortunately, I still did not understand what the problem was with the written code, I added the method `getFileExtension(Context context, Uri uri)` that gets the file extension from uri. `getType(uri)` returns not an extension, but a mimetype, which I'm using for `getExtensionFromMimeType(mimeType)`

Reply:
here's the central question: can `getExtensionFromMimeType(mimeType)` ever return `"md"`?

the code is written as if it can, but I want to make sure that's the case.

Reply:
Yes, of course I checked!

Reply:
Well, should I add a parameter `boolean isMarkdownEnabled` to `importPlaintextFile(String exportContents)` method?

Reply:
yeah I think so, or possibly better a string or enum `WithMarkdown | WithoutMarkdown` to avoid dangling and curious `true` values. I don't foresee any other options we'd want to pass in there such as tags or share urls so it should be good to introduce this one flag

Reply:
Good, did! It seemed to me, that enum will be better than a string